lib/spack/spack/cmd/env.py

```diff
@@ -523,6 +524,34 @@ def env_revert(args):
     tty.msg(msg.format(manifest_file))
 
 
+def env_generate_makefile_setup_parser(subparser):
+    pass
+
+
+def env_generate_makefile(args):
+    env = ev.active_environment()
+    if not env:
+        tty.die("required env")
+    hash_to_deps_hashes = dict()
+    hash_to_spec = dict()
+    for (_, spec) in env.concretized_specs():
+        for s in spec.traverse(root=True):
+            hash_to_spec[s.dag_hash()] = s
+            hash_to_deps_hashes[s.dag_hash()] = ' '.join(
+                [dep.dag_hash() for dep in s.dependencies()])
+
+    print(".PHONY: all clean\n")
+    print("all: {}\n".format(' '.join(
+        [s.dag_hash() for (_, s) in env.concretized_specs()])))
```

Comment:
Shouldn't `all:` just be the roots?

Reply:
they are

Reply:
maybe concretized_specs should be called concretized_root_specs